### PR TITLE
[DSS-152] Link Component

### PR DESCRIPTION
**Pull request by ju-Skinner**
# Description

Please include a summary of the change and which issue is fixed.
Please also include relevant motivation and context.
List any new dependencies or updates that are required for this change.

Fixes #(issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you've added and run to verify your changes.
Provide instructions, so we can reproduce.
Please also list any relevant details for your test configuration.

- [ ] unit tests
- [ ] e2e tests
- [ ] accessibility tests
- [ ] tested manually
- [ ] other:

**Test Configuration**:

- sage versions:
- os:
- browsers:
- screen readers:
- misc:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes

### CODE REVIEW

File: libs/core/src/components/sage-link/test/sage-link.e2e.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    await page.setContent('<sage-link href="#some-bookmark-link"></sage-link>');
+    const component = await page.find('sage-link');
+    expect(component).toHaveClass('hydrated');
+
+    const element = await page.find('sage-link >>> a');
+    expect(element.textContent).toEqual(`#some-bookmark-link`);
+
+    component.setProperty('href', '#my-new-bookmark');
+    await page.waitForChanges();
+    expect(element.textContent).toEqual(`#my-new-bookmark`);
```
Comment (goodwinchris) [inline libs/core/src/components/sage-link/test/sage-link.e2e.ts:24]: Does this test the content of the `href=""`?

Comment (QuintonJason) [reply libs/core/src/components/sage-link/test/sage-link.e2e.ts:24]: Resolved via the conversation with Julian and Chris

**Review by goodwinchris — APPROVED** [review]
Approving as an MVP 🚀

**Review by QuintonJason — APPROVED** [review]
Good to go!

**Pull request merged**